src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -829,6 +829,10 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                     return Failed;
                 }
             }
+            catch (IOException e)
```

Comment:
>catch (IOException e) [](start = 12, length = 21)

It feels like the fix is too far from the place where exception happens and it probably doesn't cover scenarios when emit is invoked through an API. We might want to intercept this within emit and always surface this as diagnostics, rather than dumping something to console.